src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -141,19 +169,27 @@ public TimeSpan(int days, int hours, int minutes, int seconds)
         /// </exception>
         public TimeSpan(int days, int hours, int minutes, int seconds, int milliseconds, int microseconds)
         {
-            long totalMicroseconds = (((long)days * 3600 * 24 + (long)hours * 3600 + (long)minutes * 60 + seconds) * 1000 + milliseconds) * 1000 + microseconds;
-            if (totalMicroseconds > MaxMicroSeconds || totalMicroseconds < MinMicroSeconds)
+            long totalMicroseconds = (days * MicrosecondsPerDay)
```

Comment:
>totalMicroseconds [](http://example.com/codeflow?start=17&length=17)

do we need to fix the possible overflow here?

Reply:
Potentially, but I tried to keep this to non-behavioral changes.

I'd prefer to put a bug fix in its own PR where it can be isolated out and have relevant tests added.